UI/window-basic-main.cpp

```diff
@@ -1603,6 +1621,314 @@ void OBSBasic::UpdateMultiviewProjectorMenu()
 			SLOT(OpenMultiviewProjector()));
 }
 
+void OBSBasic::SaveScreenshotHandler()
+{
+	const bool noSpace = true;
+
+	// Calculate output file path
+	std::string strOutputFilePath =
+		config_get_string(Config(), "ScreenShots", "FilePath");
+
+	char lastChar = strOutputFilePath.back();
+	if (lastChar != '/' && lastChar != '\\')
+		strOutputFilePath += "/";
+
+	// TODO: Take this out to configuration
+	strOutputFilePath += GenerateSpecifiedFilename("png", noSpace, "%CCYY-%MM-%DD %hh-%mm-%ss");
+
+	replace(strOutputFilePath.begin(), strOutputFilePath.end(), '\\', '/');
+
+	// Get & ensure directory exists
+	size_t last = strOutputFilePath.rfind('/');
+	if (last == string::npos)
+		return;
+
+	string directory = strOutputFilePath.substr(0, last);
+	os_mkdirs(directory.c_str());
+
+	// Save screenshot to specified file path
+	bool success = SaveScreenshotToFile(strOutputFilePath.c_str());
+
+	if (success)
+	{
+		// TODO: Audio/visual feedback
+	}
+}
+
+///
+// Use libavcodec to save an input RGBA buffer to PNG
+//
+// @param outputFilePath		PNG file path to write to
+// @param width				image_data_ptr buffer width (pixels)
+// @param height			image_data_ptr buffer height (pixels)
+// @param image_data_ptr		RGBA buffer containing image data
+// @param image_data_ptr_buffer_length	image_data_ptr data length
+// @param image_data_linesize		line size (pitch)
+//
+// @returns				true on success, otherwise false
+//
+// @note				used by OBSBasic::SaveScreenshotToFile
+//
+static bool SaveRGBABufferToPNGFile(
+	const char *outputFilePath,
+	uint32_t    width,
+	uint32_t    height,
+	uint8_t*    image_data_ptr,
+	size_t      image_data_buffer_length,
+	uint32_t    image_data_linesize)
+{
```

Comment:
This indentation of the parameters is inconsistent with the rest of the project.  It should be more like this:

```cpp
static bool SaveRGBABufferToPNGFile(
		const char  *outputFilePath,
		uint32_t    width,
		uint32_t    height,
		uint8_t     *image_data_ptr,
		size_t      image_data_buffer_length,
		uint32_t    image_data_linesize)
{
```